src/AstSemanticChecker.cpp

```diff
@@ -1032,26 +1028,27 @@ static const std::vector<SrcLocation> usesInvalidWitness(AstTranslationUnit& tu,
                 newVariableName << "+aggr_var_" << numReplaced++;
 
                 // Keep track of which variables are bound to aggregators
-                aggregatorVariables.insert(newVariableName.str());
+                variables.insert(newVariableName.str());
 
                 return std::make_unique<AstVariable>(newVariableName.str());
+            } else if (dynamic_cast<AstNegation*>(node.get())) {
+                // replace all negation terms with a boolean constraint of 'true'
```

Comment:
Comment should explain *why* (and maybe *how*, if tricky) not *what*.

Reply:
updated comments